src/directory.router.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+const restify = require('restify');
+const {sendHttpError} = require('./errors');
+const FindsProfiles = require('./users/FindsProfiles');
+const LoginsUsers = require('./users/LoginsUsers');
+const CreatesUsers = require('./users/CreatesUsers');
+const ChangesPasswords = require('./users/ChangesPasswords');
+const AddsAliases = require('./users/AddsAliases');
+
+const nonemptyString = str => (typeof str === 'string') && (str.length > 0);
+const validateUserId = nonemptyString;
+const validatePassword = pwd => nonemptyString(pwd) && (pwd.length >= 8);
+const validateAlias = alias => (typeof alias === 'object') && (!!alias)
+  && nonemptyString(alias.type) && nonemptyString(alias.value)
+  && (!alias.hasOwnProperty('public') || (typeof alias.public === 'boolean'));
+const validateAliases = aliases => Array.isArray(aliases) && aliases.every(validateAlias);
+
+const badUserId = next => sendHttpError(next, new restify.BadRequestError('Invalid User ID'));
+const badPassword = next => sendHttpError(next, new restify.BadRequestError('Password must be at least 8 characters long'));
+const badAliases = next => sendHttpError(next, new restify.BadRequestError('Some of the aliases are invalid'));
+
```

Comment:
... and if there are other calls that can fail for multiple reasons like those, same idea: would be nice to drop a line one the README.md concerning those calls, about which `restCode` could be expected.